### PR TITLE
Replace experimental flag with engine choice

### LINKED ISSUE #50778: Better setting for enabling new search
Something like `MB_SEARCH_ENGINE=fulltext`, pointing to old search by default.

### PR DESCRIPTION
**Pull request by crisptrutski**
Now that we're getting close to releasing the new search code, we want to stabilize the config. At the same time we want to future proof it for additional engines being added in the future.

Closes https://github.com/metabase/metabase/issues/50778

### CODE REVIEW

File: src/metabase/search/appdb/core.clj [hunk 1]
```diff
@@ -23,8 +24,9 @@
 
 (set! *warn-on-reflection* true)
 
-(defmethod search.engine/supported-engine? :search.engine/fulltext [_]
-  (and (public-settings/experimental-fulltext-search-enabled)
+(defmethod search.engine/supported-engine? :search.engine/appdb [_]
+  (and (or (not config/is-prod?)
+           (= "appdb" (some-> (public-settings/search-engine) name)))
```
Comment (crisptrutski) [inline src/metabase/search/appdb/core.clj:29]: I don't trust settings enforce the namespace

File: src/metabase/public_settings.clj [hunk 2]
```diff
@@ -1064,21 +1064,21 @@ See [fonts](../configuring-metabase/fonts.md).")
   :export?    true
   :type       :integer)
 
-(defsetting experimental-fulltext-search-enabled
-  (deferred-tru "Enables search engines which are still in the experimental stage")
+(defsetting search-engine
+  (deferred-tru "Which engine to use when performing search. Supported values are :in-place and :appdb")
```
Comment (crisptrutski) [inline src/metabase/public_settings.clj:1068]: It would be nice if we could have declarative validation

**Review by sansolovyov — APPROVED** [review]

File: src/metabase/search/appdb/core.clj (hunk 1 above)
Comment (piranha) [reply src/metabase/search/appdb/core.clj:29]: not sure I understand the code and the comment at the same time :)

Comment (crisptrutski) [reply src/metabase/search/appdb/core.clj:29]: I mean it would be nice for the setting to just return `:search.engine/appdb`

**Pull request merged**